physical/raft/raft_test.go

```diff
@@ -156,10 +156,49 @@ func compareDBs(t *testing.T, boltDB1, boltDB2 *bolt.DB, dataOnly bool) error {
 }
 
 func TestRaft_Backend(t *testing.T) {
-	b, dir := GetRaft(t, true, true)
-	defer os.RemoveAll(dir)
+	t.Parallel()
+	testCases := map[string]string{
+		"use raft wal": "true",
+		"use boltdb":   "false",
+	}
```

Comment:
nit: it took me about 30 seconds of looking at this to understand it! Why have two params that say the same thing in opposite ways? Oh because key is test name while value is the config setting. I generally find it easier although a little more verbose to use a slice of structs so that it's clear just from the declaration what they mean without having to check with the use-case. This also means it's much easier to add additional variations later if that is ever needed without restructuring the loop etc.

```go
testCase := []struct{
  name string
  useWAL bool
}{
  {
    name: "raft wal",
    useWal: true,
  },
  {
    name: "boltdb",
    useWal: false,
  },
}
```

If this feels like overkill, I might have gone for just `_, useWAl := range []bool{ true, false }` and generated the name with `fmt.Sprintf("wal=$q", useWal)`.

Reply:
I've used slices of structs in the past. I think in my head, this was one less struct field to deal with, but I can see how I picked the wrong test to experiment with this idea 😆. I switched it to slices of structs.